datumaro/components/extractor.py

```diff
@@ -359,6 +360,69 @@ def as_polygon(self):
     def get_area(self):
         return 0
 
+
+@attrs
+class Cuboid3d(Annotation):
+    _type = AnnotationType.cuboid_3d
+    _points = attrib(type=list, default=None)
+    label = attrib(converter=attr.converters.optional(int),
+        default=None, kw_only=True)
+
+    @_points.validator
+    def _points_validator(self, attribute, points):
+        if points is None:
+            points = [0, 0, 0,  0, 0, 0,  1, 1, 1]
+        else:
+            assert len(points) == 3 + 3 + 3, points
+            points = [round(p, _COORDINATE_ROUNDING_DIGITS) for p in points]
+        self._points = points
+
+    # will be overridden by attrs, then will be overridden again by us
+    # attrs' method will be renamed to __attrs_init__
+    def __init__(self, position, rotation=None, scale=None, **kwargs):
+        assert len(position) == 3, position
+        if not rotation:
+            rotation = [0] * 3
+        if not scale:
+            scale = [1] * 3
+        kwargs.pop('points', None)
+        self.__attrs_init__(points=[*position, *rotation, *scale], **kwargs)
+    __actual_init__ = __init__ # save pointer
+
+    @property
+    def position(self):
+        """[x, y, z]"""
+        return self._points[0:3]
+
+    @position.setter
+    def _set_poistion(self, value):
+        self.position[:] = \
+            [round(p, _COORDINATE_ROUNDING_DIGITS) for p in value]
+
+    @property
+    def rotation(self):
+        """[rx, ry, rz]"""
+        return self._points[3:6]
+
+    @rotation.setter
+    def _set_rotation(self, value):
+        self.rotation[:] = \
+            [round(p, _COORDINATE_ROUNDING_DIGITS) for p in value]
+
+    @property
+    def scale(self):
+        """[sx, sy, sz]"""
+        return self._points[6:9]
+
+    @scale.setter
+    def _set_scale(self, value):
+        self.scale[:] = \
+            [round(p, _COORDINATE_ROUNDING_DIGITS) for p in value]
+
+assert not hasattr(Cuboid3d, '__attrs_init__') # hopefully, it will be supported
+setattr(Cuboid3d, '__attrs_init__', Cuboid3d.__init__)
+setattr(Cuboid3d, '__init__', Cuboid3d.__actual_init__)
```

Comment:
Seems like you can accomplish the desired behavior without manual manipulation by using `init=False` (see https://www.attrs.org/en/stable/init.html#hooking-yourself-into-initialization).

Reply:
It was added in newer versions of attrs, so it needs a general refactoring. I've opened an issue for this: #310